envoy/http/query_params.h

```diff
@@ -15,6 +19,22 @@ namespace Utility {
 using QueryParams = std::map<std::string, std::string>;
 using QueryParamsVector = std::vector<std::pair<std::string, std::string>>;
 
+class QueryParamsMulti {
+private:
+  absl::btree_map<std::string, std::vector<std::string>> data;
+
+public:
+  void remove(absl::string_view key);
+  void add(absl::string_view key, const std::string& value);
```

Comment:
Sorry, I probably was not very clear. What I was thinking is here interface is `void add(absl::string_view key, const absl::string_view value);` and when we need to store the string, we copy it to extend its lifetime (Please see my comment below:)

Same for override() case